sdk/identity/identity/src/credentials/msalBrowser/msalAuthCode.ts

```diff
@@ -0,0 +1,265 @@
+
+    if (options.authenticationRecord) {
+      this.app.setActiveAccount(options.authenticationRecord);
+    }
+  }
+
+  /**
+   * Loads the account based on the result of the authentication.
+   * If no result was received, tries to load the account from the cache.
+   * @param result - Result object received from MSAL.
+   */
+  private async handleResult(
+    result?: msalBrowser.AuthenticationResult
+  ): Promise<AuthenticationRecord | undefined> {
+    try {
+      if (result && result.account) {
+        logger.info(`MSAL Browser V2 authentication successful.`);
+        this.app.setActiveAccount(result.account);
+        return result.account;
+      }
+
+      // If by this point we happen to have an active account, we should stop trying to parse this.
+      const activeAccount = this.app.getActiveAccount();
+      if (activeAccount) {
+        return activeAccount;
+      }
+
+      // If we don't have an active account, we try to activate it from all the already loaded accounts.
+      const accounts = this.app.getAllAccounts();
+      if (accounts.length > 1) {
+        // If there's more than one account in memory, we force the user to authenticate again.
+        // At this point we can't identify which account should this credential work with,
+        // since at this point the user won't have provided enough information.
+        // We log a message in case that helps.
+        logger.info(
+          [
+            "More than one account was found authenticated for this Client ID and Tenant ID.",
+            "However, no `authenticationRecord` has been provided for this credential,",
+            "therefore we're unable to pick between these accounts.",
+            "A new login attempt will be requested, to ensure the correct account is picked.",
+            "To work with multiple accounts for the same Client ID and Tenant ID, please provide an `authenticationRecord` when initializing `InteractiveBrowserCredential`."
+          ].join("\n")
+        );
+        // To safely trigger a new login, we're also ensuring the local cache is cleared up for this MSAL object.
+        // However, we want to avoid kicking the user out of their authentication on the Azure side.
+        // We do this by calling to logout while specifying a `onRedirectNavigate` that returns false.
+        await this.app.logout({
+          onRedirectNavigate: () => false
+        });
+        return;
+      }
+
+      // If there's only one account for this MSAL object, we can safely activate it.
+      if (accounts.length === 1) {
+        this.app.setActiveAccount(accounts[0]);
+        return accounts[0];
+      }
+
+      logger.info(`No accounts were found through MSAL.`);
+    } catch (e) {
+      logger.info(`Failed to acquire token through MSAL. ${e.message}`);
+    }
+    return;
+  }
+
+  /**
+   * Uses MSAL to handle the redirect.
+   */
+  public async handleRedirect(): Promise<AuthenticationRecord | undefined> {
```

Comment:
Might be better to not make this `public` to encourage people to call `authenticate()` on page load instead of `handleRedirect`. It may cause unintended side effects if someone uses both APIs.

Reply:
Oh sorry! This is public in this file, but we don't ever expose nothing out of the `msalBrowser` folder.

Reply:
You can see the public API changes in the file [sdk/identity/identity/review/identity.api.md](https://github.com/Azure/azure-sdk-for-js/pull/13263/files#diff-819ab338fa1a26458e9fbdd72075d0d237cddf6ff6b9248ed590404d354972a8L7).

Reply:
Gotcha, I missed that file.